src/OpenTelemetry.AutoInstrumentation/Instrumentations/Kafka/Integrations/ConsumerConsumeSyncIntegration.cs

```diff
@@ -0,0 +1,96 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Diagnostics;
+using OpenTelemetry.AutoInstrumentation.CallTarget;
+using OpenTelemetry.AutoInstrumentation.DuckTyping;
+using OpenTelemetry.AutoInstrumentation.Instrumentations.Kafka.DuckTypes;
+using OpenTelemetry.Context.Propagation;
+
+namespace OpenTelemetry.AutoInstrumentation.Instrumentations.Kafka.Integrations;
+
+/// <summary>
+/// Kafka consumer sync consume instrumentation
+/// </summary>
+[InstrumentMethod(
+    assemblyName: IntegrationConstants.ConfluentKafkaAssemblyName,
+    typeName: IntegrationConstants.ConsumerTypeName,
+    methodName: IntegrationConstants.ConsumeSyncMethodName,
+    returnTypeName: IntegrationConstants.ConsumeResultTypeName,
+    parameterTypeNames: new[] { ClrNames.Int32 },
+    minimumVersion: IntegrationConstants.MinVersion,
+    maximumVersion: IntegrationConstants.MaxVersion,
+    integrationName: IntegrationConstants.IntegrationName,
+    type: InstrumentationType.Trace)]
+public static class ConsumerConsumeSyncIntegration
+{
+    internal static CallTargetState OnMethodBegin<TTarget>(TTarget instance, int timeout)
+    {
+        // Attempting to consume another message marks the end
+        // of a previous message processing
+        KafkaCommon.StopCurrentConsumerActivity();
+
+        return CallTargetState.GetDefault();
+    }
+
+    internal static CallTargetReturn<TResponse> OnMethodEnd<TTarget, TResponse>(TTarget instance, TResponse response, Exception? exception, in CallTargetState state)
+    {
+        IConsumeResult? consumeResult;
+        if (exception is not null && exception.TryDuckCast<IResultException>(out var resultException))
+        {
+            consumeResult = resultException.ConsumerRecord;
+        }
+        else
+        {
+            consumeResult = response == null ? null : response.DuckAs<IConsumeResult>();
+        }
+
+        if (consumeResult is null)
+        {
+            return new CallTargetReturn<TResponse>(response);
+        }
+
+        var propagatedContext = Propagators.DefaultTextMapPropagator.Extract(default, consumeResult, KafkaCommon.MessageHeaderValueGetter);
+        string? spanName = null;
+        if (!string.IsNullOrEmpty(consumeResult.Topic))
+        {
+            spanName = $"{consumeResult.Topic} {MessagingAttributes.Values.ProcessOperationName}";
+        }
+
+        spanName ??= MessagingAttributes.Values.ProcessOperationName;
+        var activity = KafkaCommon.Source.StartActivity(spanName, ActivityKind.Consumer, propagatedContext.ActivityContext);
```

Comment:
It is now required to link to the propagated context (instead of parenting).

I'd recommend to link to the extracted context, and parent with the currently activated context, if there's any.

Reply:
changed in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/3055/commits/fb2e35bf6a6b09871bec0195dfb2943655521699

Reply:
Yes, we didn't want to give a clear recommendation for this in the specification.

However, when you think that in your case it's beneficial for the user experience, it's perfectly fine to set the remote context as parent (in addition to setting it as link). I would only do this when there's no current activity.